### DISCUSSION TITLE
Why can't I run the Linux program?

### DISCUSSION THREAD

**Discussion by kyuranger** [AvaloniaUI/Avalonia#11967] [Q&A] [answered]
I followed this tutorial and build a Linux program:https://docs.avaloniaui.net/docs/next/guides/platforms/rpi/running-on-Raspbian-lite-via-drm

In spite I met another program that I asked:https://github.com/AvaloniaUI/Avalonia/discussions/11950

But I found a possible solution in:
https://github.com/AvaloniaUI/Avalonia/blob/master/samples/ControlCatalog.NetCore/Program.cs
and
https://stackoverflow.com/questions/75952900/error-when-trying-to-run-avalonia-through-drm-in-raspberry-pi-4

Now I modify the code like this and build success:
```
[STAThread]
        public static int Main(string[] args)
        {
            var builder = BuildAvaloniaApp();
            double GetScaling()
            {
                var idx = Array.IndexOf(args, "--scaling");
                if (idx != 0 && args.Length > idx + 1 &&
                    double.TryParse(args[idx + 1], NumberStyles.Any, CultureInfo.InvariantCulture, out var scaling))
                    return scaling;
                return 1;
            }
            if (args.Contains("--drm"))
            {
                SilenceConsole();
                DefaultPropertyHelper.IsDRM = true;
                return builder.StartLinuxDrm(args: args, card: "/dev/dri/card1", scaling: GetScaling());
            }

            return builder.StartWithClassicDesktopLifetime(args);
        }
```

When I continue the next step in tutorial (3.3 Run app on Raspberry), the program cannot run:
![image](https://github.com/AvaloniaUI/Avalonia/assets/30720036/db0d9f1b-4ff8-455d-ad02-994cbbfaf519)

Here is the struct of the directory:
![image](https://github.com/AvaloniaUI/Avalonia/assets/30720036/548d039a-70a7-46ca-9afa-6725f391a173)

Here is my profile setting in VS:
![image](https://github.com/AvaloniaUI/Avalonia/assets/30720036/4f2dd886-364f-4d5c-b12d-d8cfb00a074b)

Here are files when I publish target to windows platform:
![image](https://github.com/AvaloniaUI/Avalonia/assets/30720036/d3fc931d-56ed-489b-8be3-4f40669885a1)

And also, in step 1.4 Verify DRM, the raspberry can render the cube successfully.

I am not sure whether it is the problem with StartLinuxDrm of my code.

I am a new beginner of Linux, please give me a help.

**Comment by hez2010** [comment]
It is not an Avalonia issue. Your program didn't run because you didn't pass the current directory to the command line. 

Instead of `sudo ProtoMark.TIJ --drm`, you need to execute `sudo ./ProtoMark.TIJ --drm`.

**Reply by kyuranger** [reply]
![image](https://github.com/AvaloniaUI/Avalonia/assets/30720036/c344116f-5954-4aca-bba2-f7303031c6c2)
It still is not working.

**Reply by hez2010** [reply]
I see that your executable file doesn't have executable permission.
Execute `chmod +x ./ProtoMark.TIJ` before running.

**Reply by kyuranger** [reply]
I tried what you said, I don't know which path I should input that I make some test:
![image](https://github.com/AvaloniaUI/Avalonia/assets/30720036/24512171-1eb5-452d-ae67-91c6f86d2efa)
The problem still here.

And here is files in the directory:
![image](https://github.com/AvaloniaUI/Avalonia/assets/30720036/587f443f-94a6-4cc6-a1bb-922ea1abd801)

The color of the file changed to green, it seems now it has executable permission but still cannot run successfullyy.

**Comment by kyuranger** [comment, marked as answer]
Problem solved. I installed a 64-bit Raspbian Lite and running a Linux arm self-contained program on it.

After I compiled the program with Linux arm64, it all worked well.
